website/releases/release-0.12.0.md

```diff
@@ -0,0 +1,132 @@
+and the development of this feature can be tracked in [HUDI-4500](https://issues.apache.org/jira/browse/HUDI-4500).
+:::
+
+### Post Write Termination Strategy with Deltastreamer
+
+Users can now configure a post write termination strategy with deltastreamer `continuous` mode if need be. For instance,
+users can configure graceful shutdown if there is no new data from source for 5 consecutive times. Here is the interface
+for the termination strategy.
+```java
+/**
+ * Post write termination strategy for deltastreamer in continuous mode.
+ */
+public interface PostWriteTerminationStrategy {
+
+  /**
+   * Returns whether deltastreamer needs to be shutdown.
+   * @param scheduledCompactionInstantAndWriteStatuses optional pair of scheduled compaction instant and write statuses.
+   * @return true if deltastreamer has to be shutdown. false otherwise.
+   */
+  boolean shouldShutdown(Option<Pair<Option<String>, JavaRDD<WriteStatus>>> scheduledCompactionInstantAndWriteStatuses);
+
+}
+```
+
+Also, this might help in bootstrapping a new table. Instead of doing one bulk load or bulk_insert leveraging a large
+cluster for a large input of data, one could start deltastreamer on continuous mode and add a shutdown strategy to 
+terminate, once all data has been bootstrapped. This way, each batch could be smaller and may not need a large cluster 
+to bootstrap data. We have one concrete implementation out of the box, [NoNewDataTerminationStrategy](https://github.com/apache/hudi/blob/0d0a4152cfd362185066519ae926ac4513c7a152/hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/NoNewDataTerminationStrategy.java).
+Users can feel free to implement their own strategy as they see fit.
+
+### Performance Improvements
+
+While some performance regressions were addressed in [0.11.1 release](/releases/release-0.11.1#addressing-performance-regression-in-0110), 
+this version brings more improvements to make Hudi the most performant lake storage format. Some notable improvements are:
+- Closed the performance gap in writing through Spark datasource vs sql. Previously, datasource writes were faster.
+- All built-in key generators implement more performant Spark-specific APIs.
+- Replaced UDF in bulk insert operation with RDD transformation to cut down serde cost.
+- Optimized column stats index performance in data skipping.
+
+We recently benchmarked Hudi against TPC-DS workload.
+Please check out [our blog](/blog/2022/06/29/Apache-Hudi-vs-Delta-Lake-transparent-tpc-ds-lakehouse-performance-benchmarks) for more details.
+
+### SQL Support Improvements
+
+- Support for upgrade, downgrade, bootstrap, clean, rollback and repair through `Call Procedure` command.
+- Support for `analyze table`.
+- Support for `Create/Drop/Show/Refresh Index` syntax through Spark SQL.
+
+### Flink Integration Improvements
+
+- Flink 1.15.x is integrated, use profile param `-Pflink1.15` when compiling the codes to adapt the version. Alternatively, use `hudi-flink1.15-bundle`.
```

Comment:
i'd put spark 3.3 support and flink 1.15 support in early sections since these are big features to highlight. in general i'd rank features from major impact to minor

Reply:
Added a separate section for Spark 3.3 and Flink 1.15 after the first top 3 features.